gridcapa-swe-runner-app/src/main/java/com/farao_community/farao/swe/runner/app/dichotomy/DichotomyParallelization.java

```diff
@@ -6,75 +6,72 @@
-import com.farao_community.farao.swe.runner.app.services.CneFileExportService;
 import com.farao_community.farao.swe.runner.app.services.OutputService;
-import com.farao_community.farao.swe.runner.app.services.VoltageCheckService;
-import org.slf4j.Logger;
-import org.slf4j.MDC;
 import org.springframework.stereotype.Service;
 
+import java.util.ArrayList;
+import java.util.List;
 import java.util.Optional;
+import java.util.concurrent.CompletableFuture;
+import java.util.concurrent.ExecutionException;
 
 /**
  * @author Theo Pascoli {@literal <theo.pascoli at rte-france.com>}
  */
 @Service
 public class DichotomyParallelization {
-    private final Logger businessLogger;
     private final DichotomyLogging dichotomyLogging;
-    private final DichotomyRunner dichotomyRunner;
     private final OutputService outputService;
-    private final VoltageCheckService voltageCheckService;
-    private final CgmesExportService cgmesExportService;
-    private final CneFileExportService cneFileExportService;
 
-    public DichotomyParallelization(Logger businessLogger, DichotomyLogging dichotomyLogging, DichotomyRunner dichotomyRunner,
-                                    OutputService outputService, VoltageCheckService voltageCheckService,
-                                    CgmesExportService cgmesExportService, CneFileExportService cneFileExportService) {
-        this.businessLogger = businessLogger;
+    private final DichotomyParallelizationWorker worker;
+
+    public DichotomyParallelization(DichotomyLogging dichotomyLogging, OutputService outputService, DichotomyParallelizationWorker worker) {
+
         this.dichotomyLogging = dichotomyLogging;
-        this.dichotomyRunner = dichotomyRunner;
         this.outputService = outputService;
-        this.voltageCheckService = voltageCheckService;
-        this.cgmesExportService = cgmesExportService;
-        this.cneFileExportService = cneFileExportService;
+        this.worker = worker;
     }
 
     public SweResponse launchDichotomy(SweData sweData) {
-        final ExecutionResult<SweDichotomyResult> executionResult = ParallelExecution
-                .of(() -> runDichotomyForOneDirection(sweData, DichotomyDirection.ES_FR))
-                // .and(() -> runDichotomyForOneDirection(sweData, Direction.FR_ES))
-                // .and(() -> runDichotomyForOneDirection(sweData, Direction.ES_PT))
-                // .and(() -> runDichotomyForOneDirection(sweData, Direction.PT_ES))
-                .close();
+        ExecutionResult<SweDichotomyResult> executionResult = runAndGetSweDichotomyResults(sweData);
         dichotomyLogging.logEndAllDichotomies();
         String ttcDocUrl = outputService.buildAndExportTtcDocument(sweData, executionResult);
         String voltageEsFrZipUrl = outputService.buildAndExportVoltageDoc(DichotomyDirection.ES_FR, sweData, executionResult);
+        outputService.buildAndExportVoltageDoc(DichotomyDirection.FR_ES, sweData, executionResult);
```

Comment:
Could you move the export of voltage file to the dichotomyParallelizationWorker?

Reply:
of course, done :)